.github/workflows/ci_cd.yml

```
@@ -45,6 +45,19 @@ jobs:
       pull-requests: write
     runs-on: ubuntu-latest
     steps:
+      - name: "Setup python"
+        uses: actions/setup-python@v2
```

Outdated version. Latest is:

```suggestion
        uses: actions/setup-python@v5
```